### PR TITLE
Improvements to the structure of stokes_waves_K.H

### PR DESCRIPTION
**Pull request by mpolimeno**
## Summary

Improving the structure of stokes_waves_K.H.
This PR builds on the improvements introduced in https://github.com/Exawind/amr-wind/commit/30fc9a15a65670ebe3226b85e76b2155b4e4251d for the computation of the free surface profile, and updates the formulation for the velocity components within the context of Stokes Waves. The major change introduced here is the refactoring of the velocity computation following Kinnas's notes https://www.caee.utexas.edu/prof/kinnas/ce358/oenotes/kinnas_stokes11.pdf based on Fenton's https://johndfenton.com/Papers/Fenton90b-Nonlinear-wave-theories.pdf

## Pull request type

Please check the type of change introduced:

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [x] new unit-test(s)
- [ ] new regression test(s)
- [ ] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [x] on CPU 
- the regression tests
  - [ ] on GPU 
  - [x] on CPU 

## Additional background

Issue Number:

### CODE REVIEW

**Comment by mpolimeno** [comment]
The basic more "cosmetic" changes are essentially done. Need to refactor the functions that compute the horizontal and vertical velocity components a bit, but I will write some small unit tests for eta and the velocities themselves first.

**Comment by mpolimeno** [comment]
Wrote a small unit test for the free surface profile. I will revisit it and update it as I keep going through Fenton (https://johndfenton.com/Papers/Fenton90b-Nonlinear-wave-theories.pdf) and Kinnas (https://www.caee.utexas.edu/prof/kinnas/ce358/oenotes/kinnas_stokes11.pdf). Need to add a test for the velocity components before refactoring.

**Comment by mpolimeno** [comment]
The unit test for the velocity is meant also as a template for the refactoring of its computation in stokes_waves_K.H

**Comment by mbkuhn** [comment]
Is this ready for review, or are there some things you are still working on adding?

**Comment by mpolimeno** [comment]
> Is this ready for review, or are there some things you are still working on adding?

@mbkuhn Almost there, I think. The last round of changes to test_waves.cpp incorporates the kind of structure that I had in mind to refactor the computation of the velocity in stokes_waves_K.H. If that structure looks sensible to you, I'll go ahead and incorporate it in the header file and then marked the PR as ready for an official review.

**Comment by mpolimeno** [comment]
Reverted a few of the changes introduced in [5f27f99] for now as a temporary fix. In any case, the updated formulation is consistent and has been tested locally for all the relevant Stokes Orders. Will prioritize the refactoring of the velocity computation in the header file for the remainder of this PR.

**Comment by mpolimeno** [comment]
This should be it for now. Further improvements/changes will be the subject of a separate PR

File: unit_tests/ocean_waves/test_wave_theories.cpp [hunk 1]
```diff
@@ -10,17 +10,17 @@ class WaveTheoriesTest : public MeshTest
 
 TEST_F(WaveTheoriesTest, StokesWaves)
 {
-    amrex::Real CoeffTol = 1e-4;
+    amrex::Real coeff_tol = 1e-4;
```
Comment (mbkuhn) [inline unit_tests/ocean_waves/test_wave_theories.cpp:13]:
```suggestion
    constexpr amrex::Real coeff_tol = 1e-4;
```

File: unit_tests/ocean_waves/test_wave_theories.cpp [hunk 2]
```diff
@@ -10,17 +10,17 @@ class WaveTheoriesTest : public MeshTest
...
 
     amrex::Real wavenumber = 2.0;
-    amrex::Real waterdepth = 0.376991;
-    int StokesOrder = 5;
-    amrex::Real c0, a11, a22, b22, c2, d2, e2, a31, a33, b31, a42;
-    amrex::Real a44, b42, b44, c4, d4, e4, a51, a53, a55, b53, b55;
+    amrex::Real water_depth = 0.376991;
+    int stokes_order = 5;
```
Comment (mbkuhn) [inline unit_tests/ocean_waves/test_wave_theories.cpp:17]:
```suggestion
    constexpr int stokes_order = 5;
```

**Review by mbkuhn — APPROVED** [review]

**Pull request merged**